### PR TITLE
Create Priority field

### PR DESCRIPTION
**Pull request by C-likethis123**
Created Priority class to represent Priority.

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Priority.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    /**
+     * Constructs a {@code Priority}.
+     *
+     * @param priority A valid priority.
+     */
+    public Priority(String priority) {
+        requireNonNull(priority);
+        checkArgument(isValidPriority(priority), MESSAGE_CONSTRAINTS);
+        value = priority;
```
Comment (WilliamRyank) [inline src/main/java/seedu/address/model/person/Priority.java]: `value = priority.toLowerCase()` so we can save all datas as lower cases

File: src/test/java/seedu/address/model/person/PriorityTest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class PriorityTest {
+    private static final Priority PRIORITY_HIGH = new Priority("high");
+    private static final Priority PRIORITY_MEDIUM = new Priority("medium");
+    private static final Priority PRIORITY_LOW = new Priority("low");
```
Comment (WilliamRyank) [inline src/test/java/seedu/address/model/person/PriorityTest.java]: is this necessary? You can just reuse the constant u declared in `Priority` class

File: src/test/java/seedu/address/model/person/PriorityTest.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    public void equals_validEquals() {
+        assertTrue(PRIORITY_HIGH.equals(PRIORITY_HIGH)); //same object
+        assertTrue(PRIORITY_MEDIUM.equals(PRIORITY_MEDIUM)); //same object
+
+        assertFalse(PRIORITY_HIGH.equals(null)); //null object
+        assertFalse(PRIORITY_HIGH.equals(PRIORITY_HIGH_CAPS));
+    }
+
+    @Test
+    public void hashCodeTest() {
```
Comment (WilliamRyank) [inline src/test/java/seedu/address/model/person/PriorityTest.java]: maybe you can check the hashcode of `high` and `HIGH` is equals when you actually change the `value = priority.toLowerCase()`

File: src/test/java/seedu/address/model/person/PriorityTest.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        assertFalse(Priority.isValidPriority("12345")); // numbers only
+        assertFalse(Priority.isValidPriority("peter the 2nd")); // alphanumeric characters
+        assertFalse(Priority.isValidPriority("Capital Tan")); // with capital letters
+
+        assertTrue(Priority.isValidPriority("high")); //high is valid
+        assertTrue(Priority.isValidPriority("medium")); //medium is valid
+        assertTrue(Priority.isValidPriority("low")); //low is valid
+        assertTrue(Priority.isValidPriority("HIGH")); //case insensitive
+        assertTrue(Priority.isValidPriority("MEDIUM")); //case insensitive
+        assertTrue(Priority.isValidPriority("LOW")); //case insensitive
```
Comment (WilliamRyank) [inline src/test/java/seedu/address/model/person/PriorityTest.java]: Should use the constant that you declared??

File: src/main/java/seedu/address/model/person/Priority.java (hunk 1 above)
Comment (C-likethis123) [reply src/main/java/seedu/address/model/person/Priority.java]: Made a change

File: src/main/java/seedu/address/model/person/Patient.java [hunk 5]
```diff
@@ -9,28 +9,58 @@
...
+            return false;
+        }
+
+        Patient otherPerson = (Patient) other;
+        return otherPerson.getNric().equals(getNric())
+                && otherPerson.getName().equals(getName())
+                && otherPerson.getPhone().equals(getPhone())
+                && otherPerson.getType().equals(getType())
+                && otherPerson.getPriority().equals(getPriority());
+    }
```
Comment (WilliamRyank) [inline src/main/java/seedu/address/model/person/Patient.java]: miss `Age` for checking

Comment (C-likethis123) [reply src/main/java/seedu/address/model/person/Patient.java]: Rectified in commit "Add age in equality check"

File: src/test/java/seedu/address/model/person/PriorityTest.java (hunk 4 above)
Comment (C-likethis123) [reply src/test/java/seedu/address/model/person/PriorityTest.java]: Rectified

File: src/test/java/seedu/address/model/person/PriorityTest.java (hunk 3 above)
Comment (C-likethis123) [reply src/test/java/seedu/address/model/person/PriorityTest.java]: Changed

File: src/test/java/seedu/address/model/person/PriorityTest.java (hunk 2 above)
Comment (C-likethis123) [reply src/test/java/seedu/address/model/person/PriorityTest.java]: Changed

File: src/main/java/seedu/address/model/person/Patient.java [hunk 6]
```diff
@@ -9,28 +9,59 @@
...
+
+        if (!(other instanceof Patient)) {
+            return false;
+        }
+
+        Patient otherPerson = (Patient) other;
+        return otherPerson.getNric().equals(getNric())
+                && otherPerson.getName().equals(getName())
+                && otherPerson.getPhone().equals(getPhone())
+                && otherPerson.getType().equals(getType())
```
Comment (SimNee) [inline src/main/java/seedu/address/model/person/Patient.java]: very minor: maybe can switch line 62 to 59. ie: the type, nric,phone then age then priority

File: src/test/java/seedu/address/model/person/PriorityTest.java [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+        // invalid type
+        assertFalse(Priority.isValidPriority("")); // empty string
+        assertFalse(Priority.isValidPriority(" ")); // spaces only
+        assertFalse(Priority.isValidPriority("^")); // only non-alphanumeric characters
+        assertFalse(Priority.isValidPriority("peter*")); // contains non-alphanumeric characters
+
+        assertFalse(Priority.isValidPriority("12345")); // numbers only
+        assertFalse(Priority.isValidPriority("peter the 2nd")); // alphanumeric characters
+        assertFalse(Priority.isValidPriority("Capital Tan")); // with capital letters
```
Comment (SimNee) [inline src/test/java/seedu/address/model/person/PriorityTest.java]: should we test ```"highlow"``` ```medium high"``` to return false?

File: src/main/java/seedu/address/model/person/Patient.java (hunk 6 above)
Comment (TheRealRavager) [reply src/main/java/seedu/address/model/person/Patient.java]: agree with simnee on this, would be better to follow attribute order

File: src/main/java/seedu/address/model/person/Priority.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+ * The higher the priority, the more likely they will be matched with donor's organs.
+ * Guarantees: immutable; is valid as declared in {@link #isValidPriority(String)}
+ */
+public class Priority {
+
+    public static final String MESSAGE_CONSTRAINTS = "Priority can only be high, medium or low"
+            + "and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
```
Comment (TheRealRavager) [inline src/main/java/seedu/address/model/person/Priority.java]: Change `address` to `priority`

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 9]
```diff
@@ -56,91 +60,106 @@
...
+                + PHONE_DESC_BOB + PHONE_DESC_AMY, new AddCommand(expectedPerson));
+
+        // multiple nrics - last nric accepted
+        assertParseSuccess(parser, TYPE_DESC_AMY + NRIC_DESC_BOB + NRIC_DESC_AMY
+                + NAME_DESC_AMY + PHONE_DESC_AMY, new AddCommand(expectedPerson));
+
+        // multiple types - last type accepted
+        assertParseSuccess(parser, TYPE_DESC_BOB + TYPE_DESC_AMY + NRIC_DESC_AMY
+                + NAME_DESC_AMY + PHONE_DESC_AMY, new AddCommand(expectedPerson));
+    }
```
Comment (TheRealRavager) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: Change to `expectedDoctor` ?

File: src/main/java/seedu/address/model/person/Patient.java [hunk 10]
```diff
@@ -9,28 +9,59 @@
...
+
     @Override
     public String toString() {
         final StringBuilder builder = new StringBuilder();
         builder.append(super.toString())
                 .append(" Age: ")
-                .append(getAge());
+                .append(getAge())
+                .append(" Priority: ")
+                .append(getPriority());
```
Comment (SimNee) [inline src/main/java/seedu/address/model/person/Patient.java]: small issue: extra line here

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 11]
```diff
@@ -27,20 +28,22 @@
...
      */
     @JsonCreator
     public JsonAdaptedPerson(@JsonProperty("type") String type, @JsonProperty("nric") String nric,
             @JsonProperty("name") String name, @JsonProperty("phone") String phone,
-            @JsonProperty("age") String age) {
+            @JsonProperty("age") String age, @JsonProperty("priority") String priority) {
         this.type = type;
         this.nric = nric;
         this.name = name;
         this.phone = phone;
```
Comment (SimNee) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java]: small issue: maybe remove this extra line

**Comment by TheRealRavager** [comment]
If the PR fixes a certain issue please tag it by doing `fixes #issueNumber` in the description

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java (hunk 11 above)
Comment (C-likethis123) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java]: Removed.

File: src/main/java/seedu/address/model/person/Patient.java (hunk 10 above)
Comment (C-likethis123) [reply src/main/java/seedu/address/model/person/Patient.java]: Removed

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java (hunk 9 above)
Comment (C-likethis123) [reply src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: Changed

File: src/main/java/seedu/address/model/person/Priority.java (hunk 8 above)
Comment (C-likethis123) [reply src/main/java/seedu/address/model/person/Priority.java]: Changed

File: src/test/java/seedu/address/model/person/PriorityTest.java (hunk 7 above)
Comment (C-likethis123) [reply src/test/java/seedu/address/model/person/PriorityTest.java]: Added

File: src/main/java/seedu/address/model/person/Patient.java (hunk 6 above)
Comment (C-likethis123) [reply src/main/java/seedu/address/model/person/Patient.java]: Changed

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 12]
```diff
@@ -56,91 +60,106 @@
...
-    //    @Test
-    //    public void parse_optionalFieldsMissing_success() {
-    //        // zero tags
-    //        Person expectedPerson = new PersonBuilder(AMY).build();
-    //        assertParseSuccess(parser, NAME_DESC_AMY + PHONE_DESC_AMY,
-    //                new AddCommand(expectedPerson));
-    //    }
+    @Test
+    public void parse_allFieldsPresentDoctor_success() {
+        Person expectedDoctor = new PersonBuilder(AMY).build();
```
Comment (TheRealRavager) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: now that doctor class has been implemented, shall we change this to reflect as such?

Comment (C-likethis123) [reply src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: Changed

File: src/main/java/organice/storage/JsonAdaptedPerson.java [hunk 13]
```diff
@@ -105,6 +111,7 @@ public Person toModelType() throws IllegalValueException {
 
         if (modelType.isDoctor()) {
             return new Doctor(modelType, modelNric, modelName, modelPhone);
+
```
Comment (SimNee) [inline src/main/java/organice/storage/JsonAdaptedPerson.java]: weird extra line here but minor issue

**Pull request merged**